### PR TITLE
Target JDK 1.8 and use sbt-github-actions for snapshot publish

### PR DESCRIPTION
**Pull request by mdedetrich**
This PR does 2 things, the first is it makes the project target JDK 1.8 and the second is that it makes the publish step be generated by sbt-github-actions rather than manually.

First thing to note is that even though this project is being changed to target JDK 1.8 (which is actually what we use to generate docs currently) its still perfectly possible to use the exact same binary with JDK 11+, you just need to add the newer sbt-paradox/sbt-paradox-theme versions (see https://github.com/sbt/sbt-paradox-material-theme/blob/main/src/main/paradox/getting-started.md#getting-started for more info). Note that it is actually intended that sbt-paradox-material-theme will continue targeting JDK 1.8/sbt-paradox 0.9.2 until its actually forced to update the version of sbt-paradox (i.e. sbt-paradox/sbt-paradox-theme will add some incompatible API which sbt-paradox-material-theme will need to use).

Changing the build to target JDK 1.8 was necessary in order to use sbt-github-actions to handle publishing (before this was a manual github workflow file). This is because sbt-github-actions uses the first value of `githubWorkflowJavaVersions` for publishing and since sbt-paradox 0.10.6 **ONLY** works with JDK 11+ it makes sense to both test for JDK 1.8 and use JDK 1.8 for publishing.

tl;dr
- Project was changed to target JDK 1.8 but it will have no impact if projects that want to use JDK 11+, they just need to follow [these instructions](https://github.com/sbt/sbt-paradox-material-theme/blob/main/src/main/paradox/getting-started.md#getting-started)
- sbt-github-actions will now also handle snapshot publishing, it will however not publish when tags are pushed.

After this PR is merged we can then add a `.scala-steward.conf` and add this project to scala-steward so that it has less maintenance burden

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -88,12 +91,18 @@ lazy val pekkoPlugin = project
 ThisBuild / githubWorkflowBuild := Seq(WorkflowStep.Sbt(List("test", "scripted")))
 
 ThisBuild / githubWorkflowTargetTags ++= Seq("v*")
-ThisBuild / githubWorkflowPublishTargetBranches := Seq()
-ThisBuild / githubWorkflowPublish := Seq()
+ThisBuild / githubWorkflowPublishTargetBranches := Seq(
+  RefPredicate.Equals(Ref.Branch("main")))
```
Comment (mdedetrich) [inline build.sbt:95]: This ensures that we only run `+publish` when a PR is merged into main, i.e. snapshots (see https://github.com/sbt/sbt-github-actions?tab=readme-ov-file#integration-with-sbt-ci-release for instructions).

File: build.sbt [hunk 2]
```diff
@@ -88,12 +91,18 @@ lazy val pekkoPlugin = project
...
+
+ThisBuild / githubWorkflowPublish := Seq(
```
Comment (mdedetrich) [inline build.sbt:97]: This is exactly the same as the original `publish.yml`.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -45,33 +45,79 @@ jobs:
...
+    steps:
+      - name: Ignore line ending differences in git
+        if: contains(runner.os, 'windows')
+        run: git config --global core.autocrlf false
 
-      - name: Setup Java (temurin@21)
-        if: matrix.java == 'temurin@21'
+      - name: Configure pagefile for Windows
+        if: contains(runner.os, 'windows')
+        uses: al-cheb/configure-pagefile-action@v1.4
```
Comment (pjfanning) [inline .github/workflows/ci.yml:90]: why all this windows stuff? not a blocker

**Review by pjfanning — APPROVED** [review]
lgtm - just a question about the windows build stuff

File: .github/workflows/ci.yml (hunk 3 above)
Comment (mdedetrich) [reply .github/workflows/ci.yml:90]: This is just making sure that the plugin can be built on windows for people such as @He-Pin . The specific code you are pointing out is a workaround for how windows handles page files (tl;dr due to limited resources of github actions runners its possible to run out of memory specifically for windows).
